pkgs/by-name/py/pyfa/package.nix

```diff
@@ -1,31 +1,124 @@
 {
   lib,
-  appimageTools,
-  fetchurl,
+  python3Packages,
+  fetchFromGitHub,
+  gsettings-desktop-schemas,
+  adwaita-icon-theme,
+  wrapGAppsHook3,
+  gdk-pixbuf,
 }:
 let
-  version = "2.60.2";
+  version = "2.61.3";
 in
-appimageTools.wrapType2 {
+python3Packages.buildPythonApplication rec {
   inherit version;
   pname = "pyfa";
+  format = "other";
 
-  src = fetchurl {
-    name = "pyfa-appimage-${version}";
-    url = "https://github.com/pyfa-org/Pyfa/releases/download/v${version}/pyfa-v${version}-linux.AppImage";
-    hash = "sha256-6doetQ6E1Occ/SqewfxRqPEX1MnUuFomm+8VmetIz4Y=";
+  src = fetchFromGitHub {
+    owner = "pyfa-org";
+    repo = "Pyfa";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-i8NcRTn817gqwQP6j0RPUJkq09eTI4nfe3EVqYnWRpo=";
   };
 
+  build-system = [ python3Packages.setuptools ];
+  dependencies = with python3Packages; [
+    wxpython
+    logbook
+    matplotlib
+    python-dateutil
+    requests
+    sqlalchemy_1_4
+    cryptography
+    markdown2
+    beautifulsoup4
+    pyaml
+    roman
+    numpy
+    python-jose
+    requests-cache
+  ];
+
+  buildInputs = [
+    gsettings-desktop-schemas
+    adwaita-icon-theme
+    gdk-pixbuf
+  ];
+
+  dontWrapGApps = true;
+  nativeBuildInputs = [
+    python3Packages.pyinstaller
+    wrapGAppsHook3
+  ];
+
+  #
+  # upstream does not include setup.py
+  #
+  patchPhase = ''
+    cat > setup.py <<EOF
+      from setuptools import setup
+      setup(
+        name = "${pname}",
+        version = "${version}",
+        scripts = ["pyfa.py"],
+        packages = setuptools.find_packages(),
+      )
+    EOF
```

Comment:
Non-blocking: You could use [`writeText`](https://nixos.org/manual/nixpkgs/unstable/#trivial-builder-writeText) here.